lib/jekyll/utils.rb

```diff
@@ -147,6 +147,14 @@ def has_yaml_header?(file)
     rescue EOFError
       false
     end
+
+    # Determine whether the given content string contains Liquid Tags or Vaiables
+    #
+    # Returns true is the string contains sequences of `{%` or `{{`
+    def has_liquid_construct?(content)
+      return false if content.nil? || content.empty?
+      content.include?("{%") || content.include?("{{")
```

Comment:
Are the strings converted to RegEx on the backend? This _might_ be faster?

```ruby
!%r!{[{%]!.match(content).nil?
```

Reply:
This *might* be faster.. or may not.. Needs to be benchmarked.. :wink: